### PR TITLE
add DGCMomentum and post-processing defenses

### PR DESCRIPTION
**Pull request by Yanghello**
添加梯度压缩优化器DGCMomentum
添加Post-Processing防御手段:

- rounding：精度裁剪
- labeling：使用argmax, 只返回label 的indice
- topk：只返回top-k值

### CODE REVIEW

File: PrivacyGuard/gradient_compress/README.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+# Privacy Preservation by Gradient Compress
+English | [简体中文](./README_cn.md)
+
+Gradient compress algorithm only uses important gradients (larger than the threshold) to update network parameters. Grandient compress not only can reduce commmunication cost in federated learning, but also can prevent gradient-based attack (such as DLG attack, see `PrivBox/inversion/dlg.py` file).
+
+DGC (Deep Gradient Compress) is one of the famous gradient compress algorithm. Detail algorithm is shown as follows [1].
```
Comment (honshj) [inline PrivacyGuard/gradient_compress/README.md:6]: Detail -> Detailed

File: PrivacyGuard/gradient_compress/README.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+<p align="center">
+  <img src="../docs/images/dgc.png?raw=true" width="700" title="DGC algorithm"/>
+</p>
+
+We implement DGC as paddle optimizer (DGCMomentum) to simple the method of usage, just as same with other paddle optimizer, such as SGD. We also give an example for DGCMomentum optimizer. Meanwhile, DLG attack is append for evaluating the defense of gradient compress.
```
Comment (honshj) [inline PrivacyGuard/gradient_compress/README.md:12]: We implement DGC as a Paddle optimizer (DGCMomentum)  just as other Paddle optimizer like SGD.

File: PrivacyGuard/post_process/labeling.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+from post_processor import PostProcessor
+
+__all__ = ["LabelingNet"]
+
+class LabelingNet(PostProcessor):
+    """
+    LabelingNet strategy changes output of network to output label indices
+    """
+
+    def __init__(self, network, axes=None, starts=None, ends=None):
```
Comment (honshj) [inline PrivacyGuard/post_process/labeling.py:29]: axes or axis?

File: PrivacyGuard/post_process/rounding.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                 precision,
+                 use_softmax=True,
+                 axes=None,
+                 starts=None,
+                 ends=None):
+        r"""
+        Rounding network output to the precision
+
+        Parameters:
+            network (Layer): trained paddle network.
```
Comment (honshj) [inline PrivacyGuard/post_process/rounding.py:40]: what's arg 'use_softmax' mean?

File: PrivacyGuard/post_process/labeling.py (hunk 3 above)
Comment (Yanghello) [reply PrivacyGuard/post_process/labeling.py:29]: paddle.slice也是用的axes，应该是因为这里可以是多个维度

**Review by honshj — APPROVED** [review]
LGTM

**Pull request merged**